src/main/scala/is/hail/variant/VariantDataset.scala

```diff
@@ -163,8 +163,10 @@ g = let
   }
 
   def summarize(): SummaryResult = {
-    vds.rdd
-      .aggregate(new SummaryCombiner[Genotype](_.hardCallIterator.countNonNegative()))(_.merge(_), _.merge(_))
+    val rowType = vds.rowType
+    vds.rdd2.mapPartitions { it =>
+      Iterator.single(it.foldLeft(new SummaryCombiner(rowType))(_.merge(_)))
+    }.reduce(_.merge(_))
```

Comment:
Is there a better pattern for doing an aggregation with per-partition allocations?